tests/Support/SupportCollectionTest.php

```diff
@@ -948,6 +948,22 @@ public function testMapWithKeys()
         );
     }
 
+    public function testMapWithKeysIntegerKeys()
+    {
+        $data = new Collection([
+            ['id' => 1, 'name' => 'A'],
+            ['id' => 3, 'name' => 'B'],
+            ['id' => 2, 'name' => 'C'],
+        ]);
+        $data = $data->mapWithKeys(function ($item) {
+            return [$item['id'] => $item];
+        });
+        $this->assertEquals(
+            [1, 3, 2],
+            $data->keys()->all()
+        );
```

Comment:
`assertSame` should be considered instead, as it does strict comparison thus also compares array orders. See [Array Operators](http://php.net/manual/en/language.operators.array.php).